From b5ebbb01b20509571eb8dab03ed7b49891d9bad4 Mon Sep 17 00:00:00 2001
From: David Edmundson <david@davidedmundson.co.uk>
Date: Sun, 13 Dec 2015 16:30:25 +0000
Subject: [PATCH 07/10] If the user runs failsafe mode, use software rendering
 in QtQuick

Graphics seem more likely than anything else to be broken, if the user
explicitly selects failsafe mode from the DM we should follow the same
rationale as kwin and not trust the user's drivers.

REVIEW: 126205
---
 startkde/startkde.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/startkde/startkde.cmake b/startkde/startkde.cmake
index 9e52fee..12c921e 100644
--- a/startkde/startkde.cmake
+++ b/startkde/startkde.cmake
@@ -6,7 +6,8 @@
 if test "x$1" = x--failsafe; then
     KDE_FAILSAFE=1 # General failsafe flag
     KWIN_COMPOSE=N # Disable KWin's compositing
-    export KWIN_COMPOSE KDE_FAILSAFE
+    QT_XCB_FORCE_SOFTWARE_OPENGL=1
+    export KWIN_COMPOSE KDE_FAILSAFE QT_XCB_FORCE_SOFTWARE_OPENGL
 fi
 
 # When the X server dies we get a HUP signal from xinit. We must ignore it
-- 
2.5.0

